pageserver/src/tenant.rs

```diff
@@ -1564,6 +1571,26 @@ impl Tenant {
         })
     }
 
+    async fn read_on_disk_heatmap(&self) -> Option<HeatMapTenant> {
```

Comment:
For an attached pageserver that restarts, the on-disk heatmap will be from whenever it was last a secondary (i.e. perhaps many days stale).

It might make sense to just have attached locations always write out the heatmap to local disk at the same time is they upload it to S3, so that all locations have a somewhat up to date heatmap on local disk irrespective of whether they were most recently secondary or attached